ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule/SinglePeer/Indices.hs

```diff
@@ -146,50 +147,59 @@ tipPointSchedule g slotLength msgDelayInterval slots = do
       let (pointSeq, newBranch) = span (\(a, b) -> a < b) xs
       times <- forM pointSeq $ \(s, _) -> do
                  delay <- uniformRMDiffTime msgDelayInterval g
-                 pure $ s + delay
+                 pure $ addTime delay s
       (times', xss) <- case newBranch of
         [] -> pure ([], [])
         ((seqLast, _) : branches) -> do
           delay <- uniformRMDiffTime msgDelayInterval g
-          let lastTime = seqLast + delay
+          let lastTime = addTime delay seqLast
           (times', xss) <- handleDelayedTipPoints lastTime branches
           pure (lastTime : times', xss)
       -- When the slots are not increasing, we must be doing a rollback.
       -- We might have tip points in past slots.
       times'' <- go xss
       pure $ times ++ times' ++ times''
 
-    -- | The start of each slot in the schedule
-    toDiffTime :: SlotNo -> DiffTime
-    toDiffTime (SlotNo s) = fromIntegral s * slotLength
+    -- | The amount of time taken by the given number of slots.
+    slotsDiffTime :: Int -> DiffTime
+    slotsDiffTime s = fromIntegral s * slotLength
 
-    -- | Assign times to tip points in past slots. A past slots is
+    -- | The time at the start of the slot.
+    slotTime :: SlotNo -> Time
+    slotTime (SlotNo s) = Time (slotsDiffTime (fromIntegral s))
+
+    -- | Assign times to tip points in past slots. A past slot is
     -- any earlier slot than the first parameter.
     --
     -- Yields the assigned times and the remaining tip points which
     -- aren't in the past.
-    handleDelayedTipPoints :: DiffTime -> [(DiffTime, DiffTime)] -> m ([DiffTime], [(DiffTime, DiffTime)])
+    handleDelayedTipPoints :: Time -> [(Time, Time)] -> m ([Time], [(Time, Time)])
     handleDelayedTipPoints lastTime xss = do
-      let (pointSeq, newBranch) = span (\(a, _) -> a + fst msgDelayInterval <= lastTime) xss
+      let (pointSeq, newBranch) = span (\(a, _) -> addTime (fst msgDelayInterval) a <= lastTime) xss
           nseq = length pointSeq
           -- The first point in xss that is not in the past
           firstLater = case newBranch of
             -- If there is no later point, pick an arbitrary later time interval
             -- to sample from
-            []           -> lastTime + toDiffTime (toEnum nseq)
-            ((a, _) : _) -> a + fst msgDelayInterval
-      times <- replicateM nseq (uniformRMDiffTime (lastTime, firstLater) g)
+            []           -> addTime (let Time lt = lastTime in lt) (slotTime (toEnum nseq)) -- REVIEW: why are we summing two absolute times?
```

Comment:
The let statement in the first argument throws me off. I think it would work better to call a function that unwraps the `DiffTime` value or introduce the let before the call to addTime.